src/data/aws_iam_types.py

```diff
@@ -46,11 +48,42 @@ class Policy:
     default_version: str
     document: Optional[Dict[str, Any]] = None
 
+    def doc_equals(self, doc: Dict[str, Any]) -> bool:
+        unrolled = self._unroll_statements(self.document) if self.document else set()
+        return unrolled == self._unroll_statements(doc)
+
+    def _unroll_statements(self, doc: Dict[str, Any]) -> Set[Statement]:
+        unrolled = set()
+        statements = doc["Statement"] if is_list(doc["Statement"]) else [doc["Statement"]]
+        for s in self._validate_statements(statements):
+            actions = s["Action"] if is_list(s["Action"]) else [s["Action"]]
+            resources = s["Resource"] if is_list(s["Resource"]) else [s["Resource"]]
+            for a in actions:
+                for r in resources:
+                    unrolled.add(Statement(action=a, resource=r, effect=s["Effect"], condition=str(s.get("Condition"))))
+        return unrolled
+
+    @staticmethod
+    def _validate_statements(statements: Sequence[Dict[str, Any]]) -> Sequence[Dict[str, Any]]:
+        unsupported_elements = ["NotAction", "NotResource", "Principal", "NotPrincipal"]
+        invalid_statements = [s for s in statements if any(element in s for element in unsupported_elements)]
+        if invalid_statements:
+            raise UnsupportedPolicyDocumentElement(f"one of {unsupported_elements} found in {invalid_statements}")
```

Comment:
What does this mean in practive? is the policy recreated?

Reply:
The policy won't be recreated. This is highlighting that the policy in use and/or the desired policy in the configuration can't be understood by the policy document comparator as they contain elements that make comparing a lot more complicated. The scanning task will fail and scanner config should be revisited!